spec/routes/web/inference_playground_spec.rb

```diff
@@ -0,0 +1,75 @@
+      lb2 = LoadBalancer.create_with_id(private_subnet_id: ps.id, name: "dummy-lb-2", src_port: 80, dst_port: 80, health_check_endpoint: "/up")
+      InferenceEndpoint.create_with_id(name: "ie1", model_name: "e5-mistral-7b-it", project_id: project_wo_permissions.id, is_public: true, visible: true, location: "loc", vm_size: "size", replica_count: 1, boot_image: "image", storage_volumes: [], engine_params: "", engine: "vllm", private_subnet_id: ps.id, load_balancer_id: lb.id)
+      InferenceEndpoint.create_with_id(name: "ie2", model_name: "e5-mistral-8b-it", project_id: project_wo_permissions.id, is_public: true, visible: false, location: "loc", vm_size: "size", replica_count: 1, boot_image: "image", storage_volumes: [], engine_params: "", engine: "vllm", private_subnet_id: ps.id, load_balancer_id: lb.id)
+      InferenceEndpoint.create_with_id(name: "ie3", model_name: "llama-guard-3-8b", project_id: project_wo_permissions.id, is_public: false, visible: true, location: "loc", vm_size: "size", replica_count: 1, boot_image: "image", storage_volumes: [], engine_params: "", engine: "vllm", private_subnet_id: ps.id, load_balancer_id: lb.id)
+      ie = InferenceEndpoint.create_with_id(name: "ie4", model_name: "llama-3-1-405b-it", project_id: project.id, is_public: false, visible: true, location: "loc", vm_size: "size", replica_count: 1, boot_image: "image", storage_volumes: [], engine_params: "", engine: "vllm", private_subnet_id: ps.id, load_balancer_id: lb2.id)
+      ie.associate_with_project(project)
+      InferenceEndpoint.create_with_id(name: "ie5", model_name: "test-model", project_id: project.id, is_public: false, visible: true, location: "loc", vm_size: "size", replica_count: 1, boot_image: "image", storage_volumes: [], engine_params: "", engine: "vllm", private_subnet_id: ps.id, load_balancer_id: lb.id)
+      visit "#{project.path}/inference-playground"
+
+      expect(page.title).to eq("Ubicloud - Playground")
+      expect(page).to have_no_content("ie1")
+      expect(page).to have_no_content("ie2")
+      expect(page).to have_no_content("ie3")
+      expect(page).to have_no_content("ie5")
+      expect(page).to have_select("inference_endpoint", selected: "ie4")
+    end
+
+    it "gives choice of inference tokens" do
+      visit "#{project.path}/inference-token"
+      expect(ApiKey.all).to be_empty
+      click_button "Create Token"
+
+      visit "#{project.path}/inference-playground"
+
+      expect(page.title).to eq("Ubicloud - Playground")
+      expect(page).to have_select("inference_token", selected: ApiKey.first.ubid)
+    end
+  end
+
+  describe "feature disabled" do
+    before do
+      project.set_ff_inference_ui(false)
+      login(user.email)
+      visit "#{project.path}/inference-playground"
+    end
+
+    it "inference playground page is not accessible" do
+      expect(page.status_code).to eq(404)
+    end
+  end
```

Comment:
since it's a single test, we can merge them
```suggestion
  describe "feature disabled" do
    it "inference playground page is not accessible" do
      project.set_ff_inference_ui(false)
      login

      visit "#{project.path}/inference-playground"

      expect(page.status_code).to eq(404)
    end
  end
```